pkg/frontend/session.go

```diff
@@ -1997,6 +1999,153 @@ func (tcc *TxnCompilerContext) Resolve(dbName string, tableName string) (*plan2.
 	return tcc.getTableDef(ctx, table, dbName, tableName)
 }
 
+func (tcc *TxnCompilerContext) ResolveUdf(name string, args []*plan.Expr) (string, error) {
+	var rowIndex int64
+	var rds []engine.Reader
+	var expectInvalidArgErr bool
+	var expectedInvalidArgLengthErr bool
+	var badValue string
+
+	dbName, err := tcc.ensureDatabaseIsNotEmpty(catalog.MO_CATALOG)
+	if err != nil {
+		return "", err
+	}
+	ctx, table, err := tcc.getRelation(dbName, "mo_user_defined_function")
+	if err != nil {
+		return "", err
+	}
+
+	expr := getUdfNameMatchExpr(name)
+	ret, err := table.Ranges(ctx, expr)
+	if err != nil {
+		return "", err
+	}
+
+	// init reader logic
+	switch {
```

Comment:
Can we use sql to execute this query? The read logic of pipeline may be modified, and it is better to generate a plan or pipeline (or even sql) to read the data, which is more reasonable. Otherwise, there will be a need to modify this code when the logic of the pipeline is modified.

Reply:
Thanks for the advice! I just changed the logic to use sql to execute the query :) and it's indeed better in many ways.